src/beamlime/applications/base.py

```diff
@@ -242,24 +177,38 @@ class Application(LogMixin):
 
     """
 
+    @dataclass
+    class Stop:
+        """A message to break the routing loop."""
+
+        content: Optional[Any]
+        sender: type
+        receiver: type
```

Comment:
Does it need these attributes? The protocol requires them. But does is that necessary? It seems to me that if you have distinct message types that don't inherit from a common base, they should be able to set their own contents freely.

Reply:
Yeah... I was not sure about the ``MessageProtocol``, how it should be
so I tried to keep it very general and flexible...

Reply:
Depends on the usage. I see several different approaches that can work. I don't know beamlime well enough to judge which one is best. But using protocols like you did does not seem like the best approach to me.

1. Single message type
```python
class Message:
    content: Any  # contains the actual message
    sender: type
    receiver: type
```
This is useful if there are fields (`sender`, `receiver`) that are needed by the router or all handlers. The message type is then represented by the type of data in `content`.

2. Use a base class
```python
class Message:
    sender: type
    receiver: type

class DataMessage(Message):
    data: DataArray
```
This is also useful if there are fields (`sender`, `receiver`) that are needed by the router or all handlers. If effectively inlines the concrete messages into the object but is otherwise the same as 1. Handlers would inspect the type of the message where they would inspect `content` in 1.

This could also be done with a protocol but that would require repeating the common fields.

3. No commonality
```python
class DataMessage:
    data: DataArray
    sender: type
    receiver: type

class StopMessage:
   ...
```
This is useful if there is no need for common attributes. In that case, handlers anyway have to check the type (or use duck-typing) and only look at the attributes they care about.

Reply:
I would like to keep it as protocol so that the handlers or daemons don't have to depend on beamlime,
probably plugin packages...? Or we can maybe change it later... 
but I did remove the ``sender`` and ``receiver`` for now.